js/src/name/publish.js

```diff
@@ -0,0 +1,96 @@
+/* eslint-env mocha */
+'use strict'
+
+const each = require('async/each')
+const hat = require('hat')
+
+const { fixtures } = require('./utils')
+const { getDescribe, getIt, expect } = require('../utils/mocha')
+
+module.exports = (createCommon, options) => {
+  const describe = getDescribe(options)
+  const it = getIt(options)
+  const common = createCommon()
+
+  describe('.name.publish', function () {
+    this.timeout(50 * 1000)
+
+    const keyName = hat()
+    let ipfs
+    let nodeId
+    let keyId
+
+    before(function (done) {
+      this.timeout(60 * 1000)
+
+      common.setup((err, factory) => {
+        expect(err).to.not.exist()
+
+        factory.spawnNode((err, node) => {
+          expect(err).to.not.exist()
+
+          ipfs = node
+          ipfs.id().then((res) => {
+            expect(res.id).to.exist()
+
+            nodeId = res.id
+
+            ipfs.key.gen(keyName, { type: 'rsa', size: 2048 }, (err, key) => {
+              expect(err).to.not.exist()
+              expect(key).to.exist()
+              expect(key).to.have.property('name', keyName)
+              expect(key).to.have.property('id')
+
+              keyId = key.id
+              populate()
+            })
+          })
+        })
+      })
+
+      function populate () {
+        each(fixtures.files, (file, cb) => {
+          ipfs.files.add(file.data, { pin: false }, cb)
+        }, done)
+      }
+    })
+
+    after((done) => common.teardown(done))
+
+    it('name publish should publish correctly', (done) => {
+      const value = fixtures.files[0].cid
+
+      ipfs.name.publish(value, true, '1m', '10s', 'self', (err, res) => {
+        expect(err).to.not.exist()
+        expect(res).to.exist()
+        expect(res).to.equal(nodeId)
```

Comment:
When you publish in go-ipfs you get back the mapping of name => value. This is also what the spec says here: https://github.com/ipfs/interface-ipfs-core/blob/master/SPEC/NAME.md#namepublish

Reply:
I was only doing this on the top level and not here. I will change this 😄

Reply:
Two other things regarding those specs.

First, the docs [spec](https://ipfs.io/docs/api/#apiv0namepublish) for publish, have the `Name` and `Value` capitalized, while the `NAME.md` spec hasn't. Should we uniform this? Go uses capitalized as well.

Second, those SPECs for [resolve](https://github.com/ipfs/interface-ipfs-core/blob/master/SPEC/NAME.md#nameresolve), they are not consistent with the docs [spec](https://ipfs.io/docs/api/#apiv0nameresolve).

Basically, the resolve should return an object with the `Path` as the docs spec have, instead of the `NAME.md` spec from this repo, right? I can also change that spec here.